src/category_theory/preadditive/endo_functor.lean

```diff
@@ -0,0 +1,53 @@
+/-
+Copyright (c) 2022 Julian Kuelshammer. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Julian Kuelshammer
+-/
+
+import category_theory.preadditive.default
+import category_theory.endofunctor.algebra
+import category_theory.preadditive.additive_functor
+
+/-!
+# Preadditive structure on algebras over a monad
+
+If `C` is a preadditive categories and `F` is an additive endofunctor on `C` then `algebra F` is
+also preadditive.
+-/
+
+universes v₁ u₁ -- morphism levels before object levels. See note [category_theory universes].
+
+namespace category_theory
+variables (C : Type u₁) [category.{v₁} C] [preadditive C] (F : C ⥤ C)
+  [functor.additive (F : C ⥤ C)]
+
+open category_theory.limits preadditive
+
+/-- The category of algebras over an additive endofunctor on a preadditive category is preadditive.
+-/
+@[simps]
+instance endofunctor.algebra_preadditive : preadditive (endofunctor.algebra F) :=
+{ hom_group := λ A₁ A₂, { add := λ α β,
+    { f := α.f + β.f,
```

Comment:
I encourage you to also provide the `nsmul` and `zsmul` fields. They will give much better defeqs when using this instance.